libs/klv/src/main/java/org/codice/ddf/libs/klv/data/text/KlvEncodingDetectedString.java

```diff
@@ -47,14 +47,17 @@ protected void decodeValue(Klv klv) {
     CharsetDetector charsetDetector = new CharsetDetector();
     charsetDetector.setText(bytes);
     CharsetMatch charsetMatch = charsetDetector.detect();
-    try {
-      value = new String(bytes, charsetMatch.getName());
-      return;
-    } catch (UnsupportedEncodingException e) {
-      LOGGER.trace(
-          "Unsupported encoding of %s, falling back to default encoding", charsetMatch.getName());
+
+    if (charsetMatch != null) {
+      try {
+        value = new String(bytes, charsetMatch.getName());
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.trace(
+            "Unsupported encoding of %s, falling back to default encoding", charsetMatch.getName());
+      }
+    } else {
+      value = new String(bytes, Charset.defaultCharset());
```

Comment:
Return after line 53.
then remove else block and 58 so that `value = ...` is always called

Reply:
Another possible solution

```java
  @Override
  protected void decodeValue(Klv klv) {
    byte[] bytes = klv.getValue();
    CharsetDetector charsetDetector = new CharsetDetector();
    charsetDetector.setText(bytes);
    CharsetMatch charsetMatch = charsetDetector.detect();
    Charset charset = Charset.defaultCharset();
    if (charsetMatch != null) {
      try {
        charset = Charset.forName(charsetMatch.getName());
      } catch (IllegalCharsetNameException e) {
        LOGGER.trace(
            "Unsupported encoding of %s, falling back to default encoding", charsetMatch.getName());
      }
    }
    value = new String(bytes, charset);
  }
```

Reply:
@rzwiefel I'll use your solution